### PR TITLE
Add icons for all the browsers in the compatibility table

### LINKED ISSUE #5460: Suggestion: Add browser icons to compatibility table
# Summary

As a UI/UX designer, I see it's better to put browser icon on browser compatibility, and colors, green and red  
because no one wants to read to see if the code works in the specific browser or not 
image

### PR DESCRIPTION
**Pull request by danielhjacobs**
## Summary

Fixes #5460

### Problem

In the redesign the browser icons were removed from the compatibility table.

### Solution

Add them to the table, copying them from the dinocons repo to this repo. ~~I need someone to help me fix the Samsung Internet symbol, which doesn't work as a mask image.~~ The svg itself looks fine, it just doesn't work right as a mask, probably because it uses the color white instead of the white part being transparent: https://github.com/danielhjacobs/yari/blob/dc147085d12c6637f6d6a3cc6edbe20cbc5ca2a5/client/src/assets/icons/samsunginternet.svg

---

### Before

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#browser_compatibility
Screenshot from 2022-03-23 11-01-54

### After

http://localhost:3000/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#browser_compatibility
Screenshot from 2022-03-23 11-01-56

### CODE REVIEW

**Comment by danielhjacobs** [comment]
Samsung Internet SVG edited with Inkscape using Path>Difference (information found from this question: https://inkscape.org/forums/questions/make-a-transparent-hole-in-a-shape/)

New look:
Screenshot from 2022-03-23 14-35-24

**Comment by schalkneethling** [comment]
@danielhjacobs Thank you for your pull request! You can use the icons [here](https://github.com/mdn/mdn-dinocons/tree/main/browsers) and [here](https://github.com/mdn/mdn-dinocons/tree/main/platforms). Those are the icons we used previously.

**Comment by schalkneethling** [comment]
@danielhjacobs Looks like you might have already done so?

**Comment by schalkneethling** [comment]
Thanks, @danielhjacobs. I am reviewing at the moment.

File: client/src/document/ingredients/browser-compatibility-table/index.scss [hunk 1]
```diff
@@ -99,6 +99,9 @@
   writing-mode: vertical-rl;
   transform: rotate(180deg);
   white-space: nowrap;
+  left: calc(50% - 0.5rem);
+  position: relative;
+  padding-top: 6px;
```
Comment (schalkneethling) [inline client/src/document/ingredients/browser-compatibility-table/index.scss:104]:
```suggestion
  padding-top: 0.5rem;
```

**Comment by schalkneethling** [comment]
I wonder what you think. I feel like giving the icon a bit of a bottom margin helps it feel a little more breezy 😄 

Something as little as:

```css
margin-bottom: 0.3rem;
```

Screenshot showing BCD table header with a bit of bottom margin added to the icons

File: client/src/assets/icons/deno.svg [hunk 2]
```diff
@@ -0,0 +1 @@
...9.002-.233-1.77-.418-3.333-.418-3.48 0-.258.295-.357.418-.148zm70.971 5.14l-.086 5.14-.308.456c-.7 1.033-.664 1.23-.627-4.526l.025-5.275.455-.468c.246-.245.492-.455.541-.455.037 0 .037 2.312 0 5.129zM75.58 79.885c.062 7.144.05 7.698-.147 7.92-.123.134-.234.22-.27.184-.087-.099-.21-15.68-.111-15.766.049-.049.172-.073.283-.061.16.037.196.91.245 7.723zM35.415 74.99c.086.763.393 4.415.516 6.137.074 1.156.074 1.193-.197 1.279-.147.049-.32.036-.368-.025-.087-.086-.357-2.816-.69-7.022l-.085-1.008h.381c.357 0 .381.024.443.64zm-9.666 4.304c.086.086.147.394.147.701 0 .32.148 1.98.308 3.702.467 4.71.504 5.35.295 5.226-.32-.184-.603-.467-.541-.54.024-.05-.025-.665-.123-1.366a61.075 61.075 0 01-.246-2.582c-.05-.714-.16-2.017-.258-2.89-.234-2.152-.222-2.398.049-2.398a.58.58 0 01.369.147zm5.817 8.276c.036.271.11 1.28.16 2.239.085 1.844.049 1.98-.456 1.598-.147-.11-.246-.578-.393-1.98-.258-2.533-.27-2.434.209-2.385.357.037.406.086.48.528z" fill="#000"/><path d="M27.017 42.156c-1.275.58-1.373 2.4-.171 3.044.87.464 1.752.245 2.206-.555.858-1.47-.54-3.173-2.035-2.49zm8.31 1.238c-1.176.761-1.176 2.593 0 3.354 1.165.736 2.673-.232 2.673-1.703 0-1.419-1.544-2.373-2.672-1.65z" fill="currentColor"/></svg>
```
Comment (schalkneethling) [inline client/src/assets/icons/deno.svg:1]: This one is missing an empty line at the end of the file.

**Review by schalkneethling — COMMENTED** [review]
Two small things. Let me know when this is ready to review again. Thank you.

File: client/src/document/ingredients/browser-compatibility-table/index.scss (hunk 1 above)
Comment (danielhjacobs) [reply client/src/document/ingredients/browser-compatibility-table/index.scss:104]: rem is the root font-size, and the root has a font-size of 16px. To use equivalent spacing, it needs to be 0.375rem

**Review by schalkneethling — APPROVED** [review]
Awesome! Thanks so much for this @danielhjacobs 🎉

**Pull request merged**